subprojects/ide/src/main/java/org/gradle/plugins/ide/eclipse/model/EclipseClasspath.java

```diff
@@ -241,11 +248,21 @@ public void setDownloadJavadoc(boolean downloadJavadoc) {
      * See {@link #file(Action)}.
      */
     public XmlFileContentMerger getFile() {
-        return file;
+        return file.getOrNull();
     }
 
     public void setFile(XmlFileContentMerger file) {
-        this.file = file;
+        this.file.set(file);
+    }
+
+    /**
+     * See {@link #setFile(XmlFileContentMerger)}.
+     *
+     * @since 4.9
+     */
+    @Incubating
+    public void setFile(Provider<XmlFileContentMerger> file) {
```

Comment:
This model object should simply own the `XmlFileContentMerger` and create the instance in its constructor.